src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -1597,22 +1600,25 @@ export class SearchView extends ViewPane {
 			}
 		};
 
-		let visibleMatches = 0;
+		this._visibleMatches = 0;
 
 		// Handle UI updates in an interval to show frequent progress and results
-		const uiRefreshHandle: any = setInterval(() => {
-			if (this.state === SearchUIState.Idle) {
-				window.clearInterval(uiRefreshHandle);
-				return;
-			}
+		if (!this._uiRefreshHandle) {
```

Comment:
prevents setting many refresh handles, which may refresh the tree when the search was already cancelled

Reply:
It seems like the clearInterval would have prevented that? This doesn't seem effectively different

Reply:
Unless it's possible to start a search before the previous one ends, which is suspicious